sycl/plugins/level_zero/usm_allocator.cpp

```diff
@@ -280,11 +261,17 @@ class Bucket {
   SystemMemory &getMemHandle();
   USMAllocContext::USMAllocImpl &getUsmAllocCtx() { return OwnAllocCtx; }
 
-  bool BucketHasCapacity();
+  bool CanPool();
 
 private:
   void onFreeChunk(Slab &);
+
+  // Get a slab to be used for chunked allocations.
+  // These slabs are used for allocations <= ChunkCutOff and not pooled.
   decltype(AvailableSlabs.begin()) getAvailSlab();
+
+  // Get a slab that will be used as a whole for a single allocation.
+  // These slabs are > ChunkCutOff in size and pooled.
```

Comment:
Why is that that some allocations don't get chunked?
We need to clearly define heuristics and the reasons behind them.

Reply:
USM allocations are a minimum of 64KB. Therefore allocations such that more than one fit in a USM allocation were chunked, not bigger ones.

Reply:
Again, please put some source code comment about this heuristic.